.github/workflows/ci.yml

```diff
@@ -0,0 +1,253 @@
+              with:
+                  name: html-report
+                  path: htmlcov
+
+    docs-deploy:
+        name: Deploy Documentation
+        runs-on: ubuntu-latest
+        needs:
+            - docs-lint
+        permissions:
+            contents: write
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                fetch-depth: 0
+            - uses: actions/setup-python@v5
+              with:
+                  python-version: 3.x
+                  cache: pip
+            - name: Install dependecies
+              run: |
+                  pip install --upgrade hatch uv
+            - name: Configure Git
+              run: |
+                 git config user.name github-actions
+                 git config user.email github-actions@github.com
+            - name: Publish Develop Docs
+              if: github.ref_name == 'main'
+              run: |
+                hatch run docs:deploy_develop
+            - name: Publish Develop Docs
+              if: startsWith(github.ref, 'refs/tags/')
+              run: |
+                  hatch run docs:deploy_latest ${{ github.ref_name }}
+
+    create-release:
+        name: Create Release
+        runs-on: ubuntu-latest
+        if: startsWith(github.ref, 'refs/tags/')
+        permissions:
+            contents: write
+        needs:
+            - build
+            - tests
+            - docs-deploy
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                fetch-depth: 0
+            - uses: actions/download-artifact@v4
+              with:
+                    name: artifacts
+                    path: dist
+            - name: Get latest release info
+              id: query-release-info
+              uses: release-flow/keep-a-changelog-action@v3
+              with:
+                command: query
+                version: ${{ github.ref_name }}
+            - name: Display release info
+              run: |
+                echo "Version: ${{ steps.query-release-info.outputs.version }}"
+                echo "Date: ${{ steps.query-release-info.outputs.release-date }}"
+                echo "${{ steps.query-release-info.outputs.release-notes }}"
+            - uses: ncipollo/release-action@v1
+              with:
+                artifacts: "dist/*.tar.gz,dist/*.whl"
+                body: ${{ steps.query-release-info.outputs.release-notes }}
+
+    pypi-publish:
+        name: Publish to PyPi
+        runs-on: ubuntu-latest
+        if: startsWith(github.ref, 'refs/tags/')
+        permissions:
+            id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+        needs:
+            - build
+            - tests
+            - docs-deploy
+        steps:
+            - uses: actions/download-artifact@v4
+              with:
+                name: artifacts
+                path: dist
+            - name: Publish build to PyPI
```

Comment:
To set this up: https://docs.pypi.org/trusted-publishers/

Reply:
Trusted publishing has been configured on PyPi.

![image](https://github.com/user-attachments/assets/c5dd1c62-ac0a-430d-997a-e299a5619cd6)